### PR TITLE
[DEVEX-110] Add Terraform module to deploy GitHub self hosted runners

### PR DESCRIPTION
**Pull request by Krusty93**
This code wraps the existing module to deploy a Container App job on an existing CAE, to use as GitHub self hosted runner.

The existing module is pretty simple, and this PR adds a wrapper that is almost a 1:1 map with it. However, I'd like to discuss it a little bit.

- Defining this module in the `dx` repository could be useful in order to have a single repository as a source of truth for developers
- We can assume some values and removing required variables:
  - both KeyVault and CAE references are one per project - 2 secrets and 2 CAE (IO + SelfCare). By removing the required variable, we can choose the one to use according to the `prefix` value

### CODE REVIEW

**Comment by Krusty93** [comment]
> > Defining this module in the dx repository could be useful in order to have a single repository as a source of truth for developers
> 
> I agree! The plan is to move this into dx repo in a second moment?

Yes

> 
> > We can assume some values and removing required variables:
> 
> Yep.
> 
> > both KeyVault and CAE references are one per project - 2 secrets and 2 CAE (IO + SelfCare). By removing the required > variable, we can choose the one to use according to the prefix value
> 
> This is not clear to me, I lack context probably.
> 
> One per "project" or one per subscription? We might have different subscriptions (or CSP) for the same project.

My bad, I meant subscription

File: infra/modules/github_selfhosted_runner_on_container_app_jobs/locals.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+locals {
```
Comment (gunzip) [inline infra/modules/github_selfhosted_runner_on_container_app_jobs/locals.tf:1]: Now that I understand better what you meant, I'm skeptical about hardcoding project specific values into shared modules (think about the `dx` repos as an open source general purpose project). AFAIK this are the only project specific settings so far.

**Review by gunzip — CHANGES_REQUESTED** [review]

File: infra/modules/github_selfhosted_runner_on_container_app_jobs/locals.tf (hunk 1 above)
Comment (Krusty93) [reply infra/modules/github_selfhosted_runner_on_container_app_jobs/locals.tf]: I restored previous config. I wonder if also the secret name should be hardcoded or not.
My proposal is to create an optional variable with a default value

Comment (gunzip) [reply infra/modules/github_selfhosted_runner_on_container_app_jobs/locals.tf]: I agree.

**Review by gunzip — APPROVED** [review]

**Pull request merged**